perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java

```diff
@@ -946,14 +946,30 @@ public void addMemberVo(PerunSession sess, Vo vo, Vo memberVo) throws RelationEx
 		checkParentVos(sess, vo, memberVo);
 		vosManagerImpl.addMemberVo(sess, vo, memberVo);
 
-		List<Member> memberVoMembers = perunBl.getMembersManagerBl().getMembers(sess, memberVo);
 		AttributeDefinition memberExpirationAttrDef;
 
 		try {
 			memberExpirationAttrDef = perunBl.getAttributesManagerBl().getAttributeDefinition(sess, AttributesManager.NS_MEMBER_ATTR_DEF + ":membershipExpiration");
 		} catch (AttributeNotExistsException e) {
 			throw new InternalErrorException(e);
 		}
+		// if VO just became a parent, assign memberOrganizations attribute to all its members
```

Comment:
Yep, I see what you mean. So creating a method that check whether VO is/was hierarchical and then using it in all the checks would probably be the best solution, correct?

Reply:
I think that members that were added while the VO didn't have any MemberVos wouldn't have this attribute set, no? We would have to think of a better way to decide if a VO has been hierarchical. Or do we want to check every time when adding members directly whether the VO wasn't hierarchical at some point and set the attribute if so?

Reply:
We discussed this and decided to create a new attribute very similar to memberOgranizations with the difference that VOs will never be removed from it. That way, we will be able to tell if a VO was hierarchical at some point, and we will also have information about all the VOs the member has ever come from. It will be done in a separate PR.